File: server/src/main/scala/producer/message_generator/data_generators/date_time/NowDateTimeGenerator.scala
package producer.message_generator.data_generators.date_time

import com.tools.teal.pulsar.ui.producer.v1.producer as pb

case class NowDateTimeGenerator()

object NowDateTimeGenerator:
    def fromPb(v: pb.NowDateTimeGenerator): NowDateTimeGenerator =
        NowDateTimeGenerator()

    def toPb(v: NowDateTimeGenerator): pb.NowDateTimeGenerator =
        pb.NowDateTimeGenerator()

File: ui/components/ui/ProducerSession/producer-task/message-generator/data-generators/date-time/now-date-time-generator.ts
import * as pb from '../../../../../../../grpc-web/tools/teal/pulsar/ui/producer/v1/producer_pb';

export type NowDateTimeGenerator = {
  type: 'now-date-time-generator'
};

export function nowDateTimeGeneratorFromPb(v: pb.NowDateTimeGenerator): NowDateTimeGenerator {
  return {
    type: 'now-date-time-generator'
  };
}

export function nowDateTimeGeneratorToPb(v: NowDateTimeGenerator): pb.NowDateTimeGenerator {
  const message = new pb.NowDateTimeGenerator();

  return message;
}
